### PR TITLE
Converted PLSC to hierarchical

### PR DESCRIPTION
**Pull request by x-tabdeveloping**
## Checklist for adding MMTEB dataset

Reason for dataset addition:
Converted both PLSC tasks (S2S, P2P) to hierarchical clustering. #702 

- [x] I have tested that the dataset runs with the `mteb` package.
- [x] I have run the following models on the task (adding the results to the pr). These can be run using the `mteb run -m {model_name} -t {task_name}` command.
  - [x] `sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2`
  - [x] `intfloat/multilingual-e5-small`
- [x] I have checked that the performance is neither trivial (both models gain close to perfect scores) nor random (both models gain close to random scores).
- [x] If the dataset is too big (e.g. >2048 examples), considering using `self.stratified_subsampling() under dataset_transform()`
- [x] I have filled out the metadata object in the dataset file (find documentation on it [here](https://github.com/embeddings-benchmark/mteb/blob/main/docs/adding_a_dataset.md#2-creating-the-metadata-object)).
- [x] Run tests locally to make sure nothing is broken using `make test`. 
- [x] Run the formatter to format the code using `make lint`. 
- [ ] I have added points for my submission to the [points folder](https://github.com/embeddings-benchmark/mteb/blob/main/docs/mmteb/points.md) using the PR number as the filename (e.g. `438.jsonl`).

### CODE REVIEW

**Comment by x-tabdeveloping** [comment]
I'm not sure whether the way I formulated the task makes sense.
@rafalposwiata You added the dataset initially, therefore you might know: Is the "disciplines" column hierarchically ordered or just multilabel? Or could "scientific_fields" be used as the first level and "disciplines" as the second? What's your take on this?

File: results/intfloat__multilingual-e5-small/PLSCHierarchicalClusteringP2P.json [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        0.07745825592646771,
+        0.08872949903414809,
+        0.0916604201461323,
+        0.08131082153986524,
+        0.07813838253031549,
+        0.09039119608717093,
+        0.09242363263566819,
+        0.08387202889701235
+      ],
+      "Level 3": [
```
Comment (KennethEnevoldsen) [inline results/intfloat__multilingual-e5-small/PLSCHierarchicalClusteringP2P.json:45]: I would cut from here and down

**Review by KennethEnevoldsen — APPROVED** [review]
Looks good

**Comment by rafalposwiata** [comment]
>Is the "disciplines" column hierarchically ordered or just multilabel? 
Disciplines are multilabel but for the added clustering tasks I chose only those cases where there is one discipline. 

> Or could "scientific_fields" be used as the first level and "disciplines" as the second?
Yes, "scientific_fields" could be used as the first level and "disciplines" as the second.

The entire dataset is available at https://huggingface.co/datasets/rafalposwiata/plsc

**Comment by KennethEnevoldsen** [comment]
@x-tabdeveloping will you add points for this then I believe it is ready to merge

**Comment by KennethEnevoldsen** [comment]
@x-tabdeveloping but the current approach is fine with that right? As I understand it is just does the clustering at each level?

**Comment by x-tabdeveloping** [comment]
Yes, unless the order is not fixed, and I don't know if it is (we have to check)

**Comment by KennethEnevoldsen** [comment]
Right. Once checked we can either close or merge

**Comment by x-tabdeveloping** [comment]
Nope, it's not hierarchical at all. We can maybe rephrase it as multilabel classification if we really want to, otherwise fine to leave it as flat clustering.

**Comment by KennethEnevoldsen** [comment]
Let us leave it as flat clustering
